src/vaadin-date-time-picker.html

```diff
@@ -292,18 +411,62 @@
             return;
           }
           this.__timePicker = timePicker;
+          this.__updateCustomFieldInputs();
 
           if (timePicker === defaultTimePicker) {
             // Synchronize properties to default time picker
             timePicker.placeholder = this.timePlaceholder;
             timePicker.step = this.step;
             timePicker.disabled = this.disabled;
             timePicker.readonly = this.readonly;
+            this.__syncI18n(timePicker, this, timePickerI18nProps);
           } else {
             // Synchronize properties from slotted time picker
             this.timePlaceholder = timePicker.placeholder;
             this.step = timePicker.step;
             // Don't set disabled or readonly from time picker since they're already set from date picker
+            this.__syncI18n(this, timePicker, timePickerI18nProps);
+          }
+
+          // min and max are always synchronized from parent to slotted because time picker min and max
+          // need to be dynamically set depending on currently selected date instead of simple propagation
+          this.__updateTimePickerMinMax();
+        }
+
+        __updateTimePickerMinMax() {
+          if (this.__timePicker && this.__datePicker) {
+            const dateEquals = Vaadin.DatePickerHelper._dateEquals;
+            const selectedDate = this.__parseDate(this.__datePicker.value);
+            const isMinMaxSameDay = dateEquals(this.__minDateTime, this.__maxDateTime);
+            const oldTimeValue = this.__timePicker.value;
+
+            if (this.__minDateTime && dateEquals(selectedDate, this.__minDateTime) || isMinMaxSameDay) {
+              this.__timePicker.min = this.__dateToIsoTimeString(this.__minDateTime);
+            } else {
+              this.__timePicker.min = this.__defaultTimeMinValue;
+            }
+
+            if (this.__maxDateTime && dateEquals(selectedDate, this.__maxDateTime) || isMinMaxSameDay) {
+              this.__timePicker.max = this.__dateToIsoTimeString(this.__maxDateTime);
+            } else {
+              this.__timePicker.max = this.__defaultTimeMaxValue;
+            }
+
+            // If time picker value changes due to the new min or max
+            // propagate the updated value change to custom-field (which doesn't
+            // automatically process programmatically triggered value updates)
+            if (this.__timePicker.value !== oldTimeValue) {
+              this.__triggerCustomFieldValueUpdate();
```

Comment:
Now in such a case as described in https://github.com/vaadin/vaadin-date-time-picker/pull/19#pullrequestreview-342440923 you'll still get a `value-changed` event with the invalid value immediately followed by one with the corrected one. It should ideally only emit one `value-changed` event with the right value.

Reply:
This has been fixed and the behaviour change a bit (as agreed in Slack).